sources/whippet/src/policy.rs

```diff
@@ -0,0 +1,897 @@
+                .iter()
+                .filter(|r| matches!(r, Rule::ConnectUser { .. }))
+                .collect();
+
+            connect_rules.sort_by(|a, b| b.get_priority().cmp(a.get_priority()));
+            let default_priority = connect_rules.first().map(|r| *r.get_priority());
+
+            if let Some(priority) = default_priority {
+                // Clean up all the default connect rules that don't match the highest priority rule
+                default_policy.rules.retain(|r| {
+                    if let Rule::ConnectUser { .. } = r {
+                        *r.get_priority() == priority
+                    } else {
+                        true
+                    }
+                });
+            }
+
+            default_priority
+        } else {
+            None
+        };
+
+        // Use the priority_threshold to clean up the user's connect rules, keeping only those with
+        // higher priority than the default priority rule
+        if let Some(user_policy) = &mut self.user {
+            user_policy.values_mut().for_each(|policy| {
+                let mut connect_rules: Vec<&Rule> =
+                    policy.rules.iter().filter(|r| r.is_connect()).collect();
+                connect_rules.sort_by(|a, b| b.get_priority().cmp(a.get_priority()));
+                let mut priority = connect_rules
+                    .first()
+                    .map(|r| *r.get_priority())
+                    .unwrap_or(0);
+
+                if let Some(priority_threshold) = priority_threshold {
+                    if priority_threshold > priority {
+                        priority = priority_threshold;
+                    }
+                };
+
+                policy.rules.retain(|r| {
+                    // Retain all rules except connect rules with lower priority
+                    // than the highest priority found
+                    !r.is_connect() || *r.get_priority() == priority
+                })
+            });
+
+            // Remove any user contexts without any rules
+            user_policy.retain(|_, p| !p.rules.is_empty());
+        }
+    }
+}
+
+/// Loads the policy from the provided path, setting the priorities for the loaded rules
+fn load_one_policy(
+    policy_file: impl AsRef<Path>,
+    base_policy: &mut Policy,
+    current_priority: &mut u64,
+) -> Result<()> {
+    let policy_file = policy_file.as_ref();
+    let policy_content = std::fs::read_to_string(policy_file).context(error::ReadPathSnafu {
+        what: policy_file.display().to_string(),
+    })?;
+    let mut policy: Policy = toml::from_str(&policy_content).context(error::PolicyParseSnafu)?;
+    policy.set_rule_priorities(current_priority);
+
+    if let Some(default_rules) = policy.default {
+        if let Some(default_context) = &mut base_policy.default {
+            default_context.rules.extend(default_rules.rules);
+        } else {
+            base_policy.default = Some(default_rules);
+        }
+    }
```

Comment:
Better variable names would help keep track of where we're working with the policy we just loaded, and where we're working with the base policy:
```suggestion
    if let Some(default_policy) = policy.default {
        if let Some(base_policy_default) = &mut base_policy.default {
            base_policy_default.rules.extend(default_policy.rules);
        } else {
            base_policy.default = Some(default_policy);
        }
    }
```